nrkno/sofie-package-manager
# feat: implement a pool of workers reserved for "playout-critical" work

## About Me
This pull request is posted on behalf of the NRK.


## Type of Contribution

This is a:
Feature


## Current Behavior

Playout-critical work can be stuck behind long-lasting lower-priority jobs.

## New Behavior

A number of the workers, available at any given time, can be reserved for fulfilling "critical" (`requiredForPlayout`) expectations. These workers will not be included in the auto-scaling mechanism (i.e. will work permanently and not be spun down when idle), but will be still cycled when they have worked for too long.


## Testing Instructions

* Set the `minCriticalWorkerApps` to 1 or more and verify that the playout expectations get picked up ASAP


## Other Information
This PR is based on https://github.com/nrkno/sofie-package-manager/pull/160, and that one needs to be merged first.

## Status
